src/backend/commands/matview.c

```diff
@@ -3291,23 +3291,16 @@ clean_up_IVM_hash_entry(MV_TriggerHashEntry *entry, bool is_abort)
 	{
 		MV_TriggerTable *table = (MV_TriggerTable *) lfirst(lc);
 
-		if (table->old_tuplestores)
+		list_free(table->old_tuplestores);
+		list_free(table->new_tuplestores);
+		if (!is_abort)
 		{
-			list_free(table->old_tuplestores);
-			table->old_tuplestores = NIL;
-		}
-		if (table->new_tuplestores)
-		{
-			list_free(table->new_tuplestores);
-			table->new_tuplestores = NIL;
+			ExecDropSingleTupleTableSlot(table->slot);
```

Comment:
need to add if (CurrentResourceOwner = entry->resowner), then 
`ExecDropSingleTupleTableSlot(table->slot); table_close(table->rel, NoLock);`
in case of abort error. See https://github.com/cloudberrydb/cloudberrydb/issues/552

Reply:
ok, sure, now i see where it comes from

Reply:
LGTM.